### PR TITLE
Adding clusterSize distribution

### PR DESCRIPTION
**Pull request by kgoldfeld**
This is addressing Issue #194

### CODE REVIEW

**Comment by kgoldfeld** [comment]
@assignUser I don't know if you want to take a look at this before I go ahead and merge, but I added a simple distribution called _clusterSize_. I'm thinking it will actually be useful (at least for me).

File: NEWS.md [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
 ## New features
-* Added double-dot (dynamic) functionality to defSurv. Users can now specificy 
+* Added double-dot (dynamic) functionality to defSurv. Users can now specify 
 double-dot variables in scale, shape, and formula parameters.
+* It is possible to generate variable cluster sizes using the *clusterSize*
+distribution in `defData` and `defDataAdd`.
+
+## Major fix
+* Updated the random effect variance generation for function `iccRE` under the
+Poisson distribution.
```
Comment (assignUser) [inline NEWS.md:11]: What was the issue here? Was there a bug that might have influenced existing code/results?

File: R/generate_dist.R [hunk 2]
```diff
@@ -595,3 +601,27 @@
...
+    if (d == 0) {
+      ss <- floor(total/n)
+      s <- rep(ss, n)
+      surplus <- total - ss*n
+      if (surplus > 0) {
+        upgrade <- sample(1:n, surplus, replace = FALSE)
+        s[upgrade] <- s[upgrade] + 1
+      }
+    } else {
+      x <- dirmult::rdirichlet(1, alpha = rep(1/d, n) )[1,]
```
Comment (assignUser) [inline R/generate_dist.R:622]: You added `dirmult` as a Suggests so you should guard this function to error gracefully if the package is not installed (I think we have done this in other functions too?)

File: NEWS.md (hunk 1 above)
Comment (assignUser) [reply NEWS.md:11]: Looking at the code below "improved" might be clearer  than updated?

**Review by assignUser — APPROVED** [review]
Some minor comments :)

File: NEWS.md (hunk 1 above)
Comment (kgoldfeld) [reply NEWS.md:11]: Yeah - I was trying to find the source of my code - it looked really odd (though I know I did it). I found the source that I thought I was using (it is now referenced in the iccRE.md file), and it was quite different and simpler than what I was doing. It also has the benefit of being correct. Though the ICC is not such an obvious concept in the context of the Poisson distribution, so I am not sure if the one I generated was wrong or just a different definition. This isn't great, but I just can't figure out where the old method came from.

**Comment by kgoldfeld** [comment]
@assignUser Did you see the other `clusterSize` addition, and it all looked reasonable?

**Pull request merged**